core/templates/dev/head/domain/collection/GuestCollectionProgressService.js

```diff
@@ -0,0 +1,146 @@
+// Copyright 2017 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service that records progress guests make during a collection
+ * playthrough. Note that this service does not currently support saving a
+ * user's progress when they create an account.
+ */
+
+// TODO(bhenning): Move this to a shared service which stores state across the
+// frontend in a way that can be persisted in the backend upon account
+// creation.
+
+// TODO(bhenning): This should be reset upon login, otherwise the progress will
+// be different depending on the user's logged in/logged out state.
+
+oppia.factory('GuestCollectionProgressService', [
+  function() {
+    var COLLECTION_STORAGE_KEY = 'collectionProgressStore';
```

Comment:
Should we add something like temp or v1 or some random string to the key, since this is meant to be temporary and we want to deprecate it later?

Also, might be worth having a "reset progress" button?

Reply:
Added v1, though there's no good notion of versioning here. If we want to clean this up/migrate it later, it should be simple to do.

A reset progress button is simple enough, but what's the benefit? I don't expect learners to accidentally complete an exploration, and they can always return to any exploration they wish to repeat. Do you think learners will want to experience the entire pathway again or just a single exploration?

Reply:
Entire pathway. Useful if devices need to be reset after the trial.

Reply:
(Suggest showing this only if there _is_ progress to be reset, btw.)

Reply:
So is the benefit for us to be able to reset the devices, or for users wanting to replay the collection in order? The latter seems less likely to me and is why I'm slightly reluctant to add this.

If the goal is to reset devices after the trial, we can just clear local storage on the devices themselves which will clear these temporary caches.

Reply:
Discussed offline; we're going to avoid doing this since it should be suitably easy for developers to clear local storage without a button, and we're instead going to whitelist the feature specifically to the fractions collection to avoid any user confusion that may come from state persistence.

Reply:
Just a note, let's have the whitelist be empty by default. We'll add the fractions collection id as part of the release process (I'll make a note).

Thanks!

(Oh and btw, there's another merge conflict, sorry)